frontend/src/components/Gallery.js

```diff
@@ -21,15 +24,33 @@ export default function Gallery() {
         default:
             return(
                 <GalleryStyled>
-                    {products.map(product =>
-                        <ProductPage key={product.id} productId={product.id} />
+                    {products.results.map((product, index) =>
+                        <ProductPage 
+                            key={product.id} 
+                            productId={product.id} 
+                            lastProduct={index+1 === products.results.length} 
+                            onLoadNext={handleLoadNext} 
+                            nextProductPage={!!products.nextPage} 
+                            firstProduct={index === 0} 
+                            onLoadPrev={handleLoadPrev} 
+                            prevProductPage={!!products.prevPage}
+                        />
                     )}
+                    <div ref={ref}></div>
```

Comment:
I don't understand why this ref is needed.

Reply:
it's not 😅
I forgot to delete it after testing some scrolling options...